src/base/Notifier.sol

```diff
@@ -88,6 +88,26 @@ abstract contract Notifier is INotifier {
         emit Unsubscription(tokenId, address(_subscriber));
     }
 
+    /// @dev note this function also deletes the subscriber address from the mapping
+    function _notifyBurn(uint256 tokenId, address owner, PositionInfo info, uint256 liquidity, BalanceDelta feesAccrued)
+        internal
+    {
+        ISubscriber _subscriber = subscriber[tokenId];
+
+        // remove the subscriber
+        delete subscriber[tokenId];
```

Comment:
i don't love removing this here. also it feels like something we should do before any modifyLiquidity happens at all, but then notifyBurn would have to be called before modifyLiquidity which is weird..

Reply:
if you're implying we're still violating CEI -- it might be okay since previously we were calling `_unsubscribe` (with a state change) after modifyLiquidity (interaction)

essentially were writing a special cased unsubscribe which receives additional data

Reply:
mmm yeah i think a couple of things here
1. if its going to be in here i think the function should be called `_notifyBurnAndDeleteSubscriber` or something
2. the CEI breaking ... I cant think of any problems. But worth flagging for the audit